pkg/kubectl/resource/selector.go

```diff
@@ -31,40 +32,47 @@ type Selector struct {
 	Mapping              *meta.RESTMapping
 	Namespace            string
 	LabelSelector        labels.Selector
+	FieldSelector        fields.Selector
 	Export               bool
 	IncludeUninitialized bool
 }
 
 // NewSelector creates a resource selector which hides details of getting items by their label selector.
-func NewSelector(client RESTClient, mapping *meta.RESTMapping, namespace string, selector labels.Selector, export, includeUninitialized bool) *Selector {
+func NewSelector(client RESTClient, mapping *meta.RESTMapping, namespace string, selector labels.Selector, fieldSelector fields.Selector, export, includeUninitialized bool) *Selector {
 	return &Selector{
 		Client:               client,
 		Mapping:              mapping,
 		Namespace:            namespace,
 		LabelSelector:        selector,
+		FieldSelector:        fieldSelector,
 		Export:               export,
 		IncludeUninitialized: includeUninitialized,
 	}
 }
 
 // Visit implements Visitor
 func (r *Selector) Visit(fn VisitorFunc) error {
-	list, err := NewHelper(r.Client, r.Mapping).List(r.Namespace, r.ResourceMapping().GroupVersionKind.GroupVersion().String(), r.LabelSelector, r.Export, r.IncludeUninitialized)
+	// cannot combine label selector and field selector together
```

Comment:
Why?

Reply:
Because `apiserver` does not support it.

All the `fieldSelector` and `labelSelector` are appended as query params. It is hard to differentiate them, unless we've applied new rules on this.